cli/src/main/kotlin/commands/DownloaderCommand.kt

```diff
@@ -180,15 +180,25 @@ class DownloaderCommand : CliktCommand(name = "download", help = "Fetch source c
         when (input) {
             is FileType -> {
                 val ortFile = (input as FileType).file
-                val (analyzerResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>().analyzer?.result }
+                val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
 
                 log.perf {
                     "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in " +
                             "${duration.inMilliseconds}ms."
                 }
 
-                requireNotNull(analyzerResult) {
-                    "The provided ORT result file '${ortFile.canonicalPath}' does not contain an analyzer result."
+                requireNotNull(ortResult) {
+                    "The provided ORT result file '${ortFile.canonicalPath}' has no content."
+                }
+
+                val analyzerResult = ortResult.analyzer?.result
+
+                if (analyzerResult == null) {
+                    log.warn {
+                        "The provided ORT result file '${ortFile.canonicalPath}' does not contain an analyzer result."
+                    }
+
+                    throw ProgramResult(0)
```

Comment:
You're changing the semantics here.
Before your change a result without an analyzer result would result in exit code 1.
Not this will result in a warning and the exit code 0.
What is the reason for this?

Reply:
When reviewing this code due to the changes, it seemed inappropriate to me to fail the download in that case, because if we ask to do nothing (due to a missing analyzer result) then also doing nothing is actually the right thing. However, a warning is emitted as it's rather unusual to have no analyzer result. But you're right that this change should have gone to a separate commit.

Anyway, before I do that, do you agree to the rationale?

Reply:
Yes I agree with this rationale.

Reply:
I've moved this out to a separate PR: https://github.com/oss-review-toolkit/ort/pull/3878